frontend/src/metabase/actions/containers/ActionCreator/tests/ActionCreator-Common.unit.spec.tsx

```diff
@@ -0,0 +1,76 @@
+import nock from "nock";
+import userEvent from "@testing-library/user-event";
+
+import { screen } from "__support__/ui";
+
+import {
+  createMockImplicitQueryAction,
+  createMockQueryAction,
+} from "metabase-types/api/mocks";
+
+import { setup as baseSetup, SetupOpts } from "./common";
+
+async function setup({
+  action = createMockImplicitQueryAction(),
+  ...opts
+}: SetupOpts = {}) {
+  await baseSetup({ action, ...opts });
+  return { action };
+}
+
+describe("ActionCreator > Common", () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  describe.each([
+    ["query", createMockQueryAction],
+    ["implicit", createMockImplicitQueryAction],
+  ])(`%s actions`, (_, getAction) => {
+    describe("with write permissions", () => {
+      it("should show action settings button", async () => {
+        await setup({ action: getAction(), canWrite: true });
+        const button = screen.getByRole("button", { name: "Action settings" });
+        expect(button).toBeInTheDocument();
+      });
+
+      it("should be able to set success message", async () => {
+        await setup();
+
+        userEvent.click(
```

Comment:
I want to note that, the last time I had flakes with `ActionCreator`, `userEvent` is one of the thing that make the test perform slower and eventually miss the async assertion timeout (1 second), I didn't have the exact numbers, but using it make tests run like a few hundred ms slower which is enough in some cases to miss the 1 second timeout with async assertion.

Though it might be a bad way to fix the problem, but until we can resolve the root cause (probably something with `nock` as Alex P. has said so many times), using a pure `element.click()` might help if this test ever flakes.